portfolio/src/main/webapp/repeatedElements.js

```diff
@@ -12,83 +12,134 @@
+  containerDiv.className = "container";
+  logoDiv.className = "logo";
 
-  if(inpString !== undefined) {
-    basicHeader += `<p> x ${inpString} </p>`
+  containerDiv.appendChild(logoDiv);
+  contentDiv.appendChild(containerDiv);
+
+  for (logo in LOGOSLIST) {
+    logoDiv.appendChild(generateLogo(LOGOSLIST[logo]));
   }
-  
-  document.getElementById("header").innerHTML = basicHeader + "</h1>";
 
+  document.getElementById("footer").appendChild(contentDiv);
 }
 
+
 /**
- * generate the HTML for an img at path imgPath
+ * If subTitle is the empty string keep it
+ * Else format as "x CAPS(subTitle)"
  */
-function generateTile(imgPath) {
-  const reduceString = (accum, currString) => accum + currString;
-
-  var tileHTML = ["<div class='container-box'>\
-                      <div class='tile'>\
-                        <img src=",
-                  imgPath,
-                  "     >\
-                      </div>\
-                    </div>\
-                  "];
-
-  return tileHTML.reduce(reduceString);
+function formatSubtitle(subTitle) { 
+  return (subTitle === "" ? "" : `x ${subTitle.toUpperCase()}`);
+}
+
+/**
+ * Generate header at top of doc
+ * This time with objs instead of basic strings
+ */
+function header(subTitle = "") { 
+  var headerDiv = document.createElement("h1");
+  var headerLinkDiv = document.createElement("a");
+  var subTitleDiv = document.createElement("p");
+
+  headerLinkDiv.href = "index.html";
+  
+  var headerTitle = document.createTextNode(HEADERTITLE);
+  var subTitle = document.createTextNode(formatSubtitle(subTitle));
+  
+  subTitleDiv.appendChild(subTitle);
+  headerLinkDiv.appendChild(headerTitle);
+
+  headerDiv.appendChild(headerLinkDiv);
+  headerDiv.appendChild(subTitleDiv);
+
+  document.getElementById("header").appendChild(headerDiv);
 }
 
+
 /**
  * Generate file path for some imgPrefix_i at index i in imgPrefix_images
  */
 function generateImgPath(imgPrefix, tileIndex) {
   return `images/${imgPrefix}_images/${imgPrefix}_${tileIndex}.jpeg`;
 }
 
+
+/**
+ * generate the HTML for an img at path imgPath
+ */
+function generateTile(tilePath) {
+  var containerBoxDiv = document.createElement("div");
+  var tileDiv = document.createElement("div");
+  var imgDiv = document.createElement("img");
+
+  containerBoxDiv.classname = "container-box";
+  tileDiv.className = "tile"
+  imgDiv.src = tilePath;
+
+  tileDiv.appendChild(imgDiv);
+  containerBoxDiv.appendChild(tileDiv);
+  
+  return containerBoxDiv;
+}
+
+
 /**
  * Generate a flexbox with two image-populated tiles in it
  * Prefix is some x \in [me, projects, travel, hiking]
  * i.e., twoTiles("me", 1, 2, "tiles1");
  */
 function nTiles(imgPrefix, tileIndexs, elId) {
-  var tilesHTML = "<div class='container'>";
+  var tilesDiv = document.createElement("div");
+  tilesDiv.className = "container";
+
   for (var i = 0; i < tileIndexs.length; i++) {
     tileIPath = generateImgPath(imgPrefix, tileIndexs[i]); //1-indexd
```

Comment:
why does this say 1-indexed?  You are starting from i = 0 so if it's 1-indexed would 0 be a non-value?

Reply:
old comment; forgot to remove.